File: src/main/java/no/nav/veilarbperson/client/digdir/DigdirClient.java
package no.nav.veilarbperson.client.digdir;

import lombok.SneakyThrows;
import no.nav.common.health.HealthCheck;
import no.nav.common.types.identer.Fnr;
import no.nav.veilarbperson.config.CacheConfig;
import org.springframework.cache.annotation.Cacheable;

public interface DigdirClient extends HealthCheck {

    DigdirKontaktinfo hentKontaktInfo(Fnr fnr);

    @Cacheable(CacheConfig.DIFI_HAR_NIVA_4_CACHE_NAME)
    @SneakyThrows
    HarLoggetInnRespons harLoggetInnSiste18mnd(Fnr fnr);
}

File: src/main/java/no/nav/veilarbperson/client/digdir/DigdirClient.kt
package no.nav.veilarbperson.client.digdir

import no.nav.common.health.HealthCheck

interface DigdirClient : HealthCheck {
    fun hentKontaktInfo(personidenter: KRRPostPersonerRequest): KRRPostPersonerResponse?
}

data class KRRPostPersonerRequest(
    val personidenter: Set<String>
)

data class KRRPostPersonerResponse(
    val personer: Map<String, DigdirKontaktinfo>,
    val feil: Map<String, String>?
)

data class DigdirKontaktinfo(
    val personident: String?,
    val aktiv: Boolean,
    val kanVarsles: Boolean,
    val reservasjonOppdatert: String?,
    val reservert: Boolean,
    val spraak: String?,
    val spraakOppdatert: String?,
    val epostadresse: String?,
    val epostadresseOppdatert: String?,
    val epostadresseVerifisert: String?,
    val mobiltelefonnummer: String?,
    val mobiltelefonnummerOppdatert: String?,
    val mobiltelefonnummerVerifisert: String?)




